src/main/drivers/tim_def.h

```diff
@@ -0,0 +1,357 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <platform.h>
+#include "common/utils.h"
+
+#if defined(STM32F3)
+
+#define DEF_TIM(tim, chan, pin, flags, out) {\
```

Comment:
Yes, as it happens most of the DMA is fixed. Theres only T17 and T16 that can be assigned a different DMA. The F4 theres a few that have optional streams (T1 and T8 for example - and one channel on T2). So theres not much point in allowing dynamic allocation at this stage. Perhaps for different MCUs.